src/sagas/pushNotification.js

```diff
@@ -0,0 +1,440 @@
+      });
+    }
+  } catch (error) {
+    console.error(`Error creating channel for push notification: ${error.message}`, error);
+  }
+};
+
+/**
+ * This function is called when the wallet is initialized with success.
+ */
+export function* onAppInitialization() {
+  yield take(types.START_WALLET_SUCCESS);
+
+  const { enabled, showAmountEnabled } = STORE.getItem(pushNotificationKey.settings) || { enabled: false, showAmountEnabled: false };
+  const hasBeenEnabled = STORE.getItem(pushNotificationKey.hasBeenEnabled);
+  const enabledAt = STORE.getItem(pushNotificationKey.enabledAt);
+
+  const persistedDeviceId = STORE.getItem(pushNotificationKey.deviceId);
+  const deviceId = yield call(getDeviceId);
+  // If the deviceId is different from the persisted one, we should update it.
+  // The first time the perisistedDeviceId will be null, and the deviceId will be
+  // the one returned by getDeviceId, which gets the deviceId from FCM.
+  if (deviceId && persistedDeviceId !== deviceId) {
+    STORE.setItem(pushNotificationKey.deviceId, deviceId);
+    yield put(pushUpdateDeviceId({ deviceId }));
+  }
+
+  // Initialize the pushNotification state on the redux store
+  yield put(pushInit({
+    deviceId,
+    settings: {
+      enabled,
+      showAmountEnabled,
+    },
+    hasBeenEnabled,
+    enabledAt,
+  }));
+
+  yield call(createChannelIfNotExists);
+  yield call(confirmDeviceRegistrationOnFirebase);
+  yield call(installForegroundListener);
+
+  // Check if the last registration call was made more then a week ago
+  if (hasBeenEnabled) {
+    const timeSinceLastRegistration = moment().diff(enabledAt, 'weeks');
+    if (timeSinceLastRegistration > 1) {
+      // Update the registration, as per Firebase's recommendation
+      yield put(pushRegistrationRequested({
+        enabled,
+        showAmountEnabled,
+        deviceId,
+      }));
+    }
+  }
+}
+
+/**
+ * It is responsible for loading the wallet for every push notification action that requires
+ * to interact with the wallet service api. When the user is using the facade wallet, it will
+ * load the wallet service, otherwise it will use the wallet already loaded in the redux store.
+ */
+export function* loadWallet() {
+  // This is a work-around so we can dispatch actions from inside callbacks.
+  let dispatch;
+  yield put((_dispatch) => {
+    dispatch = _dispatch;
+  });
+
+  const useWalletService = yield select((state) => state.useWalletService);
+
+  // If the user is not using the wallet-service,
+  // we need to initialize the wallet on the wallet-service first
+  let walletService;
+  if (!useWalletService) {
+    // Set urls for wallet service
+    config.setWalletServiceBaseUrl(WALLET_SERVICE_MAINNET_BASE_URL);
+    config.setWalletServiceBaseWsUrl(WALLET_SERVICE_MAINNET_BASE_WS_URL);
+    const network = new Network(NETWORK);
+
+    const pin = yield call(showPinScreenForResult, dispatch);
+    walletService = new HathorWalletServiceWallet({
+      requestPassword: pin,
```

Comment:
`requestPassword` expects a callback, not the pin. I think this should be `showPinScreenForResult`